end_to_end/end_to_end_suite_test.go

```diff
@@ -2242,6 +2242,65 @@ LANGUAGE plpgsql NO SQL;`)
 				"tplain": 1000, "ths": 1000, "tht": 1000})
 		})
 	})
+	Describe("Properly hanldes user defined type if different schema used", func() {
+		It("restores table with enum when exporting statistics", func() {
+			testhelper.AssertQueryRuns(backupConn, `create schema cdm;
+													create type cdm.status AS ENUM ('ACTIVATING', 'ACTIVATED', 'ERROR');
+													create cast (text as cdm.status) with inout as implicit;`)
+			testhelper.AssertQueryRuns(backupConn, `create table cdm.contact_status (
+															contact_id uuid not null,
+															status cdm.status not null
+														) distributed by (contact_id);
+													insert into cdm.contact_status 
+													select 
+													md5(random()::text)::uuid,
+													(array['ACTIVATING', 'ACTIVATED', 'ERROR'])[1 + x%3]
+													from generate_series(1,20000) x;
+													analyze cdm.contact_status;
+												`)
```

Comment:
Why do we need casting here? The tests work just fine without it!
```suggestion
			testhelper.AssertQueryRuns(backupConn, `create schema cdm;
													create type cdm.status as enum ('ACTIVATING', 'ACTIVATED', 'ERROR');
													create table cdm.contact_status (
														contact_id uuid not null,
														status cdm.status not null
													) distributed by (contact_id);
													insert into cdm.contact_status select
														md5(random()::text)::uuid,
														(array['ACTIVATING', 'ACTIVATED', 'ERROR'])[1 + x%3]::cdm.status
													from generate_series(1, 20000) x;`)
```

Reply:
The cast was taken from the original issue and meant to demonstrate that original query is completely functional now.

Reply:
Casting has no effect on error and reproduction!

Reply:
Moreover, even the enum has no effect on the error! The error occurred if the column type was not in search_path. So the tests can be simplified significantly.

Reply:
> The cast was taken from the original issue and meant to demonstrate that original query is completely functional now.

And if the original query included a 100500 rows, would you have included it in the tests too?

Reply:
1. You suggest to replace cast  with cast operator. A one-to-one change, one entity to another so what is the profit? I think cast here is quite in place.
2. The error happens if custom column type not in search path, right, in different schema. Enum here is an example of a custom type, and IMHO is a good example - it's readable, brings no doubts why it was used and has distinguishable values. I believe it's good here.
3. If scenario should use 100.5K rows, that's OK, no big deal. This test use 20K rows to make a statistics distinguishable, it's  fast and effective.

Reply:
Okay, then I suggest simplifying it even more.
```suggestion
			testhelper.AssertQueryRuns(backupConn, `create schema cdm;
													create type cdm.ct as (a text);
													create table cdm.t (a int, b cdm.ct) distributed by (a);
													insert into cdm.t values (1, '(1)'), (2, '(2)'), (3, '(3)');`)
```

Reply:
This simplified scenario still exposes the error, so it is sufficient for testing.